crates/bevy_ecs/src/query/access.rs

```diff
@@ -54,76 +56,115 @@ impl<T: SparseSetIndex> Access<T> {
     pub fn read_all(&mut self) {
         self.reads_all = true;
     }
 
-    /// Returns true if this `Access` has read access to all indices.
-    pub fn reads_all(&self) -> bool {
+    /// Returns `true` if the access can read all elements in the collection.
+    #[inline]
+    pub fn has_read_all(&self) -> bool {
         self.reads_all
     }
 
-    /// Clears all recorded accesses.
+    /// Sets this `Access` as having write access to all indexed elements.
+    pub fn write_all(&mut self) {
+        self.reads_all = true;
+        self.writes_all = true;
+    }
+
+    /// Returns `true` if the access can write all elements in the collection.
+    #[inline]
+    pub fn has_write_all(&self) -> bool {
+        self.writes_all
+    }
+
+    /// Removes all accesses.
     pub fn clear(&mut self) {
         self.reads_all = false;
+        self.writes_all = false;
         self.reads_and_writes.clear();
         self.writes.clear();
     }
 
-    /// Extends this `Access` with another, copying all accesses of `other` into this.
+    /// Adds all read and write access from `other`.
     pub fn extend(&mut self, other: &Access<T>) {
         self.reads_all = self.reads_all || other.reads_all;
+        self.writes_all = self.writes_all || other.writes_all;
         self.reads_and_writes.union_with(&other.reads_and_writes);
         self.writes.union_with(&other.writes);
     }
 
-    /// Returns true if this `Access` is compatible with `other`.
+    /// Returns `true` if the access and `other` can be active at the same time.
     ///
-    /// Two `Access` instances are incompatible with each other if one `Access` has a write for
-    /// which the other also has a write or a read.
+    /// `Access` instances are incompatible if one can write
+    /// an element that the other can read or write.
     pub fn is_compatible(&self, other: &Access<T>) -> bool {
-        if self.reads_all {
-            0 == other.writes.count_ones(..)
+        if self.writes_all {
+            other.reads_and_writes.count_ones(..) == 0
+        } else if other.writes_all {
+            self.reads_and_writes.count_ones(..) == 0
+        } else if self.reads_all {
+            other.writes.count_ones(..) == 0
         } else if other.reads_all {
-            0 == self.writes.count_ones(..)
+            self.writes.count_ones(..) == 0
         } else {
             self.writes.is_disjoint(&other.reads_and_writes)
                 && self.reads_and_writes.is_disjoint(&other.writes)
         }
     }
 
-    /// Calculates conflicting accesses between this `Access` and `other`.
+    /// Returns a vector of elements that the access and `other` cannot access at the same time.
     pub fn get_conflicts(&self, other: &Access<T>) -> Vec<T> {
         let mut conflicts = FixedBitSet::default();
-        if self.reads_all {
-            conflicts.extend(other.writes.ones());
+
+        if self.writes_all {
+            conflicts.extend(other.reads_and_writes.ones());
         }
```

Comment:
this also does not seem to handle `other.reads_all`

Reply:
Can you describe what you expect should happen?

Reply:
Resolving in favor of centralizing discussion under my latest comment.